single_kernel_mongo/managers/mongodb_operator.py

```diff
@@ -298,9 +299,7 @@ def install_workloads(self) -> None:
         # Truncate the file.
         self.workload.write(self.workload.paths.config_file, "")
 
-    @override
-    def prepare_for_startup(self) -> None:
-        """Handler on start."""
+    def _prepare_checks(self):
```

Comment:
I think we can rename this method, as it does quite a few things, and add a docstring + return type hint

perhaps: `_run_startup_checks`, and offload the status logic up in the call chain